### PR TITLE
List tile docs

### LINKED ISSUE #55726: Exception caught when  put ListTile  in a Row
Below is my source code,HomePage is the  file name.
If delete ListTile , everything is fine, otherwise run code with exception.
```
import 'package:flutter/material.dart';

class HomePage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        children: <Widget>[
          Row(
            children: <Widget>[
              Text('影院热映'),
              ListTile(
                title: Text('查看更多'),
                trailing: Icon(Icons.keyboard_arrow_right),
              )
            ],
          )
        ],
      ),
    );
  }
}
```
## Exception log:
```
RenderBox was not laid out: RenderSemanticsGestureHandler#4d7b3 relayoutBoundary=up8 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
'package:flutter/src/rendering/box.dart':
Failed assertion: line 1694 pos 12: 'hasSize'
The relevant error-causing widget was: 
  ListTile file:///Users/shaojunlv/AndroidStudioProjects/douban/lib/src/HomePage.dart:12:15
```

### PR DESCRIPTION
**Pull request by Calamity210**
## Description

Added a snippet to a `ListTile`'s documentation to provide an example on how to overcome a `ListTile` trying to expand without proper constraints within a `Row`.

## Related Issues

Fixes #55726

## Tests

N/A

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

This is not a breaking change.

### CODE REVIEW

File: packages/flutter/lib/src/material/list_tile.dart [hunk 1]
```diff
@@ -232,6 +232,25 @@ enum ListTileControlAffinity {
...
+///         leading: FlutterLogo(),
+///         title: Text('The ListTile is expanded to fill the available space'),
+///       ),
+///     ),
+///   ],
+/// )
+/// ```
+/// {@end-tool}
+/// {@tool snippet}
+///
```
Comment (shihaohong) [inline packages/flutter/lib/src/material/list_tile.dart:253]: Could we move this to under the first example? The first example is a generic set of use-cases for `ListTile`, and this particular API doc addresses a unique use-case.

File: packages/flutter/lib/src/material/list_tile.dart [hunk 2]
```diff
@@ -232,6 +232,25 @@ enum ListTileControlAffinity {
 ///
 /// {@tool snippet}
 ///
+/// This example uses an [Expanded] within a [Row] to demonstrate how to
+/// overcome the issue with a [ListTile] attempting to expand horizontally
+/// without proper constraints.
```
Comment (shihaohong) [inline packages/flutter/lib/src/material/list_tile.dart:237]: Just some wording recommendations:

```suggestion
/// To use a [ListTile] within a [Row], it needs to be wrapped in an
/// [Expanded] widget. [ListTile] requires fixed width constraints, 
/// whereas a [Row] does not constrain its children.
```

File: packages/flutter/lib/src/material/list_tile.dart [hunk 3]
```diff
@@ -232,6 +232,25 @@ enum ListTileControlAffinity {
...
+///
+/// ```dart
+/// Row(
+///   children: const <Widget>[
+///     Expanded(
+///       child: ListTile(
+///         leading: FlutterLogo(),
+///         title: Text('The ListTile is expanded to fill the available space'),
+///       ),
+///     ),
```
Comment (shihaohong) [inline packages/flutter/lib/src/material/list_tile.dart:247]: I would maybe add another widget or two into the `Row` before/after the `ListTile` to make the example more realistic. Having only a single `ListTile` in a `Row` does not really make sense :)

**Comment by Calamity210** [comment]
@shihaohong resolved conflicts

**Review by shihaohong — APPROVED** [review]
LGTM! Thanks for the contribution!

**Pull request merged**